WordPress/gutenberg
# Nav offcanvas restore click to edit

## What?


Removes the "edit" button for each node and allows clicking on a node to trigger "edit mode" for that block.

## Why?


See rationale in https://github.com/WordPress/gutenberg/issues/47023#issuecomment-1410148005.

Kudos to @talldan who [suggested this originally](https://github.com/WordPress/gutenberg/issues/47023#issuecomment-1378150821) and received push back 🙇‍♂️ 

## How?


- Remove dedicated edit button
- Reinstate click to edit behaviour
- Transfer a11y affordances to node
- ~Ensure only valid blocks are editable~ We decided it's better to allow all blocks to be editable and simply _imply_/hint that certain blocks are not editable via UI affordances. Ultimately if a user wants to check controls on Page List Items they should be able to. It's also better for a11y as it's more consistent to retain same behaviour across all nodes.

## Testing Instructions





- Nav block with menu that includes standard links _and_ a Page List.
- Hover over items and check there is no longer an "edit" button
- See that clicking on a node triggers edit mode.
- Test with screen reader that appropriate label is read out "Edit X".
- Test possible to edit via keyboard.
- Test possible to expand/collapse nodes without triggering edit.
- ~Test not possible to edit Page List Item blocks.~ No longer needed. See above.

### Testing Instructions for Keyboard


## Screenshots or screencast 

https://user-images.githubusercontent.com/444434/215750185-a6c293cc-9864-4cf8-b79c-3fbc0b151e7c.mp4
